File: Rails/app/models/like.rb
# frozen_string_literal: true

class Like < ApplicationRecord
  belongs_to :user
  belongs_to :submission

  validates :user_id, uniqueness: { scope: :submission_id }

  def self.liked?(user, submission)
    Like.find_by(user: user, submission: submission)
  end
end

File: Angular/src/app/models/like.ts
export class Like {
    id: number;
    userId: number;
    submissionId: number;

    constructor(id: number, user_id: number, submission_id: number) {
        this.id = id;
        this.userId = user_id;
        this.submissionId = submission_id;
    }
}
